common/sgx/collateral.c

```diff
@@ -307,17 +309,41 @@ oe_result_t oe_validate_revocation_list(
     // will error out.
     for (uint32_t i = 0; i < OE_SGX_ENDORSEMENTS_CRL_COUNT; ++i)
     {
-        OE_CHECK_MSG(
-            oe_crl_read_pem(
-                &crls[i],
-                sgx_endorsements
+        // v1/v2 CRL is PEM encoded which starts with "-----BEGIN X509 CRL-----"
+        if (strncmp(
```

Comment:
I'd suggest using `memcmp()` and checking length of the buffer to ensure it's smaller than OE_PEM_BEGIN_CRL_LEN.

Reply:
I'd suggest we also check endorsement length before memcmp(). I.e.

````
        if (sgx_endorsements
                        ->items[OE_SGX_ENDORSEMENT_FIELD_CRL_PCK_CERT + i]
                        .size >= OE_PEM_BEGIN_CRL_LEN &&
            memcmp(
                (const char*)sgx_endorsements
                    ->items[OE_SGX_ENDORSEMENT_FIELD_CRL_PCK_CERT + i]
                    .data,
                OE_PEM_BEGIN_CRL,
                OE_PEM_BEGIN_CRL_LEN) == 0)
````
It's impossible for a valid der-encoded crl to be that small, but we want to protect against malicious input.